pkg/server/plugin/upstreamauthority/wrapper.go

```diff
@@ -0,0 +1,80 @@
+package upstreamauthority
+
+import (
+	"context"
+	"crypto/x509"
+
+	"github.com/spiffe/spire/pkg/server/plugin/upstreamca"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type wrapper struct {
+	upstreamCA upstreamca.UpstreamCA
+}
+
+// Wrap creates a wrapper for UpstreamCA interface that satisfies the UpstreamAuthority interface
+func Wrap(upstreamCA upstreamca.UpstreamCA) UpstreamAuthority {
```

Comment:
Same here, why don't we use *upstreamca.UpstreamCA? Since Plugins.UpstreamCA is of this type.

Reply:
Pointers to interfaces are a code smell and unwieldy to use. The `Plugins` struct has a pointer to an interface because that is the contract with the catalog code to indicate an optional plugin field when `Fill()`'ing the struct. I'd personally prefer that smell stay contained to plugin loading code.

Reply:
same response, it is because of Fill that it is a pointer in catalog

Reply:
Oh, sorry, I didn't check that upstreamca.UpstreamCA was an interface, so now it makes sense to not have a pointer.